tests/integration_tests/functional/test_snapshot_advanced.py

```diff
@@ -25,106 +19,79 @@
     """
-    Restore from snapshots obtained with previous versions of Firecracker.
+    Restore snapshots from previous supported versions of Firecracker.
+
+    For each firecracker release:
+    1. Snapshot with the past release
+    2. Restore with the current build
 
     @type: functional
     """
+
+    # due to ARM bug fixed in commit 822009ce
+    if platform.machine() == "aarch64" and firecracker_release.version_tuple < (
+        1,
+        1,
+        4,
+    ):
+        pytest.skip("incompatible with aarch64 and Firecracker <1.1.4")
+
     # Microvm: 2vCPU 256MB RAM, balloon, 4 disks and 4 net devices.
     logger = logging.getLogger("old_snapshot_many_devices")
     builder = MicrovmBuilder(bin_cloner_path)
-
-    # With each binary create a snapshot and try to restore in current
-    # version.
-
-    firecracker.download()
-    jailer = firecracker.jailer()
-    jailer.download()
-
-    logger.info("Creating snapshot with Firecracker: %s", firecracker.local_path())
+    jailer = firecracker_release.jailer()
+    logger.info("Using Firecracker: %s", firecracker_release.local_path())
     logger.info("Using Jailer: %s", jailer.local_path())
-
-    target_version = firecracker.base_name()[1:]
-
-    # v0.23 does not support creating diff snapshots.
-    # v0.23 does not support balloon.
-    diff_snapshots = "0.23" not in target_version
-
-    # Create a snapshot.
+    diff_snapshots = True
+    logger.info("Create snapshot")
     snapshot = create_snapshot_helper(
         builder,
         logger,
         drives=scratch_drives,
         ifaces=net_ifaces,
-        fc_binary=firecracker.local_path(),
+        fc_binary=firecracker_release.local_path(),
         jailer_binary=jailer.local_path(),
         diff_snapshots=diff_snapshots,
         balloon=diff_snapshots,
     )
 
-    # Resume microvm using current build of FC/Jailer.
+    logger.info("Resume microvm using current build of FC/Jailer")
     microvm, _ = builder.build_from_snapshot(
         snapshot, resume=True, diff_snapshots=False
     )
+    logger.info("Validate all devices")
     validate_all_devices(logger, microvm, net_ifaces, scratch_drives, diff_snapshots)
     logger.debug("========== Firecracker restore snapshot log ==========")
     logger.debug(microvm.log_data)
 
 
-@pytest.mark.parametrize(
-    "firecracker",
-    firecracker_artifacts(
-        min_version="1.2.0",
-        max_version=get_firecracker_version_from_toml(),
-    ),
-    ids=firecracker_id,
-)
-def test_restore_old_version(bin_cloner_path, firecracker):
+def test_restore_current_to_old(bin_cloner_path, firecracker_release):
     """
     Restore current snapshot with previous versions of Firecracker.
 
-    Current snapshot (i.e a machine snapshotted with current build) is
-    incompatible with any past release due to notification suppression.
+    For each firecracker release:
+    1. Snapshot with the past release
+    2. Restore with the current build
```

Comment:
I think its the other way around (take snapshot with current build, load in past release)

Reply:
True. Doh!